Commit #2e2d75c breaks arm64 builds

As the title says, commit [#2e2d75c](https://github.com/LinuxCNC/linuxcnc/commit/2e2d75c8860fa65487aee78c080304f7d77e3987) breaks arm64 builds with the following output in terminal during building:

`Compiling realtime hal/drivers/hal_speaker.c
hal/drivers/hal_speaker.c:76:10: fatal error: sys/io.h: No such file or directory
   76 | #include <sys/io.h>
      |          ^~~~~~~~~~
compilation terminated.
make: *** [Makefile:1227: objects/rthal/drivers/hal_speaker.o] Error 1
`

[Phillip Carter]
> As the title says, commit #2e2d75c breaks arm64 builds with the
> following output in terminal during building:

I believe the RTAI build was already broken on arm64, as this driver was
built when RTAI was enabled.  The change only exposed the same build
failure for uspace builds.

> `Compiling realtime hal/drivers/hal_speaker.c
> hal/drivers/hal_speaker.c:76:10: fatal error: sys/io.h: No such file or directory
>    76 | #include <sys/io.h>
>       |          ^~~~~~~~~~
> compilation terminated.
> make: *** [Makefile:1227: objects/rthal/drivers/hal_speaker.o] Error 1
> `

According to
<URL: https://stackoverflow.com/questions/68300986/linux-c-sys-io-h-not-found-on-arm64-but-on-amd64-it-is-there >
this is an architectural difference issue.  I believe the solution is to
disable this driver on all non-x86 platforms, as it will only work
there.

-- 
Happy hacking
Petter Reinholdtsen

--- a/src/Makefile
+++ b/src/Makefile
@@ -63,6 +63,13 @@ TRIVIAL_BUILD=no
 endif
 endif
 
+# Used to enable x86 only HAL parts
+ifeq ($(shell uname -m),i686)
+X86ARCH = true
+endif
+ifeq ($(shell uname -m),x86_64)
+X86ARCH = true
+endif
 
 # Beautify output
 # ---------------------------------------------------------------------------
@@ -949,8 +956,10 @@ sampler-objs := hal/components/sampler.o $(MATHSTUB)
 # Subdirectory: hal/drivers
 obj-$(CONFIG_HAL_PARPORT) += hal_parport.o
 hal_parport-objs := hal/drivers/hal_parport.o $(MATHSTUB)
+ifeq ($(X86ARCH),true)
 obj-$(CONFIG_HAL_SPEAKER) += hal_speaker.o
 hal_speaker-objs := hal/drivers/hal_speaker.o $(MATHSTUB)
+endif
 ifneq ($(BUILD_SYS),uspace)
 obj-$(CONFIG_PCI_8255) += pci_8255.o
 pci_8255-objs := hal/drivers/pci_8255.o
@@ -1284,7 +1293,9 @@ endif
 ../rtlib/hal_ax5214h$(MODULE_EXT): $(addprefix objects/rt,$(hal_ax5214h-objs))
 ../rtlib/hal_ppmc$(MODULE_EXT): $(addprefix objects/rt,$(hal_ppmc-objs))
 ../rtlib/hal_skeleton$(MODULE_EXT): $(addprefix objects/rt,$(hal_skeleton-objs))
+ifeq ($(X86ARCH),true)
 ../rtlib/hal_speaker$(MODULE_EXT): $(addprefix objects/rt,$(hal_speaker-objs))
+endif
 ../rtlib/opto_ac5$(MODULE_EXT): $(addprefix objects/rt,$(opto_ac5-objs))
 ../rtlib/scope_rt$(MODULE_EXT): $(addprefix objects/rt,$(scope_rt-objs))
 ../rtlib/hal_lib$(MODULE_EXT): $(addprefix objects/rt,$(hal_lib-objs))

--- a/src/hal/drivers/hal_speaker.c
+++ b/src/hal/drivers/hal_speaker.c
@@ -4,6 +4,7 @@
 *               This file, 'hal_speaker.c', drives the PC speaker based
 *               on up to 8 bit outputs.  When the new outputs differ
 *               from the old outputs, a click is output on the speaker.
+*               This implementation only work on the x86 architecture.
 *               
 * Author: John Kasunich and Jeff Epler
 * License: GPL Version 2
@@ -60,6 +61,7 @@
     information, go to www.linuxcnc.org.
 */
 
+#include "config.h"     /* environment flags */
 #include "rtapi.h"		/* RTAPI realtime OS API */
 #include "rtapi_app.h"		/* RTAPI realtime module decls */
 #include "hal.h"		/* HAL public API decls */
@@ -68,12 +70,14 @@
    instead of rtapi_outb() and rtapi_inb() - the <asm.io> ones
    are inlined, and save a microsecond or two (on my 233MHz box)
 */
+#if defined(RTAPI_RTAI)
 #define FASTIO
+#endif /* RTAPI_RTAI */
 
 #ifdef FASTIO
 #define rtapi_inb inb
 #define rtapi_outb outb
-#include <sys/io.h>
+#include <asm/io.h>
 #endif
 
 /* module information */
@@ -155,13 +159,15 @@ int rtapi_app_main(void)
 	return -1;
     }
 
-    /* STEP 1.1: get access to port */
+#if !defined(RTAPI_RTAI)
+    /* STEP 1.1: get access to port, only needed in uspace builds */
     if (ioperm(SPEAKER_PORT, 1, 1) < 0) {
 	rtapi_print_msg(RTAPI_MSG_ERR,
 	    "SPEAKER: ERROR: ioperm() failed\n");
 	hal_exit(comp_id);
 	return -1;
     }
+#endif /* RTAPI_RTAI */
 
     /* STEP 2: allocate shared memory for skeleton data */
     port_data_array = hal_malloc(num_ports * sizeof(speaker_t));

